web/modules/custom/dpl_das/src/Plugin/rest/resource/DigitalArticleOrderResource.php

```diff
@@ -0,0 +1,163 @@
+ *
+ * Accessing this service requires a service account with an username and
+ * password and is thus not designed for direct browser based access by patron.
+ * Consequently it is exposed through the CMS instead.
+ *
+ * There is no canonical path for orders since this is not supported by the
+ * service. Patrons will have to consult their mail for confirmation that an
+ * order has been placed.
+ *
+ * @RestResource (
+ *   id = "dpl_das_digital_article_order",
+ *   label = @Translation("Digital Article Order"),
+ *   uri_paths = {
+ *     "create" = "/dpl_das/order",
+ *   },
+ *
+ *   payload = {
+ *     "name" = "order",
+ *     "description" = "Digital article order",
+ *     "in" = "body",
+ *     "schema" = {
+ *       "type" = "object",
+ *       "properties" = {
+ *         "pid" = {
+ *           "type" = "string",
+ *         },
+ *         "email" = {
+ *           "type" = "string",
+ *         },
+ *       },
+ *     },
+ *   },
+ *
+ *   responses = {
+ *     201 = {
+ *       "description" = "OK",
+ *     },
+ *     400 = {
+ *      "descriptions" = "Invalid input",
+ *     },
+ *     500 = {
+ *       "description" = "Internal server error",
+ *     },
+ *   }
+ * )
+ */
+class DigitalArticleOrderResource extends ResourceBase {
+
+  /**
+   * Serializer used to validate/convert incoming requests to value objects.
+   *
+   * @var \Symfony\Component\Serializer\SerializerInterface
+   */
+  private SerializerInterface $serializer;
+
+  /**
+   * Elba client to pass incoming requests to.
+   *
+   * @var \Drupal\dpl_das\Elba\Client
+   */
+  private Client $client;
+
+  /**
+   * Access to local configuration.
+   *
+   * @var \Drupal\Core\Config\ConfigManagerInterface
+   */
+  private ConfigManagerInterface $configManager;
+
+  /**
+   * Creates an instance of the plugin.
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The container to pull out services used in the plugin.
+   * @param mixed[] $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   *
+   * @return static
+   *   Returns an instance of this plugin.
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->getParameter('serializer.formats'),
+      $container->get('logger.factory')->get('rest'),
+    );
+
+    $instance->serializer = $container->get('dpl_das.serializer');
+    $instance->client = $container->get('dpl_das.client');
+    $instance->configManager = $container->get('config.manager');
+
+    return $instance;
+  }
+
+  /**
+   * Receive incoming requests to order digital articles.
+   */
```

Comment:
Personally I am not a fan of declaring `@throws` blocks. It is not something we require from our coding guidelines. If you like it I can add it but it may become somewhat random in regard to where we use it and where we do not. We could enforce it through code analysis if we wanted.

Based on the future of the integration do you want me to implement this?